packages/toc/src/panel.ts

```diff
@@ -28,7 +28,12 @@ export class TableOfContentsPanel extends SidePanel {
     this._title = new Private.Header(this._trans.__('Table of Contents'));
     this.header.addWidget(this._title);
 
-    this._treeview = new TableOfContentsWidget();
+    this._treeview = new TableOfContentsWidget({
+      placeholderHeadline: this._trans.__('No Headings'),
+      placeholderText: this._trans.__(
+        'The table of contents shows headings in Jupyter notebooks.'
```

Comment:
```suggestion
        'The table of contents shows headings in notebooks and supported files.'
```

Unless it changed in 4.0, In Python files ToC shows the outline (function/class definitions). It should (aspirationally) also show up in markdown files. Maybe even shorter:

`Open a notebook with headings or supported file to see outline`